CMakeLists.txt

```diff
@@ -20,180 +14,153 @@ include(GNUInstallDirs)
-# builds just link against the system vulkan library.
-if (NOT ANDROID)
-    FetchContent_Declare(
-        vulkan-loader
-        GIT_REPOSITORY ${GIT_VULKAN_LOADER_REPOSITORY}
-        GIT_TAG ${GIT_VULKAN_LOADER_COMMIT}
+    if (DEFINED CMAKE_TOOLCHAIN_FILE)
+        set(optional_args ${optional_args} "--cmake_var")
+        set(optional_args ${optional_args} "CMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+    endif()
+
+    # Add a target so that update_deps.py will run when necessary
+    # NOTE: This is triggered off of the timestamps of known_good.json and helper.cmake
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_LIST_DIR}/external/helper.cmake
+        COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/scripts/update_deps.py 
+            --dir ${CMAKE_CURRENT_LIST_DIR}/external --arch ${_target_arch} --config ${_build_type} --generator "${CMAKE_GENERATOR}" ${optional_args}
+        DEPENDS ${CMAKE_CURRENT_LIST_DIR}/scripts/known_good.json
     )
+
+    # Check if update_deps.py needs to be run on first cmake run
+    if (${CMAKE_CURRENT_LIST_DIR}/scripts/known_good.json IS_NEWER_THAN ${CMAKE_CURRENT_LIST_DIR}/external/helper.cmake)
+        execute_process(
+            COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/scripts/update_deps.py
+                --dir ${CMAKE_CURRENT_LIST_DIR}/external --arch ${_target_arch} --config ${_build_type} --generator "${CMAKE_GENERATOR}" ${optional_args}
+            WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+            RESULT_VARIABLE _update_deps_result
+        )
+        if (NOT (${_update_deps_result} EQUAL 0))
+            message(FATAL_ERROR "Could not run update_deps.py which is necessary to download dependencies.")
+        endif()
+    endif()
+    include(${CMAKE_CURRENT_LIST_DIR}/external/helper.cmake)
+else()
+    message("********************************************************************************")
+    message("* NOTE: Not adding target to run update_deps.py automatically.                 *")
+    message("********************************************************************************")
+    find_package(PythonInterp 3 QUIET)
 endif()
 
-# After all the content is declared, it can safely be made available.
-# For SDK (flat) builds, we'll receive pointers to our dependency repository builds.
-# Otherwise, we'll fetch and build ourselves.  Other repositories also respect environment
-# variables, so we're doing it here too.
-if(PROFILES_BUILD_TESTS)
-    FetchContent_MakeAvailable(googletest)
+if (VALIJSON_INSTALL_DIR)
+    list(APPEND CMAKE_PREFIX_PATH ${VALIJSON_INSTALL_DIR})
 endif()
+if (GOOGLETEST_INSTALL_DIR)
+    list(APPEND CMAKE_PREFIX_PATH ${GOOGLETEST_INSTALL_DIR})
+endif()
+if (VULKAN_HEADERS_INSTALL_DIR)
+    list(APPEND CMAKE_PREFIX_PATH ${VULKAN_HEADERS_INSTALL_DIR})
+endif()
+
+find_package(VulkanHeaders REQUIRED CONFIG QUIET)
+set(VULKAN_HEADER_REGISTRY ${CMAKE_CURRENT_SOURCE_DIR}/external/Vulkan-Headers/registry)
```

Comment:
This approach won't work to find the registry reliably.

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/9706e6962c2a32c19cdcbe68d00244a9ca15cbf5/cmake/VVLGenerateSourceCode.cmake#L21